### DISCUSSION TITLE
PHP Client Extension

### DISCUSSION THREAD

**Discussion by aszenz** [duckdb/duckdb#10377] [Ideas]
I would like to integrate duck db into a php codebase, primarily to use it as a query front-end for connecting with mysql and enjoying duckdbs rich sql support.

I found a way to use duck db with php using php ffi (available since 7.4), see:
https://github.com/thbley/php-duckdb-integration, but it requires modifying the headers file to remove preprocessors and needs manual type mapping. 

Is anyone else interested in a duckdb extension for php or someone working on it

**Comment by aszenz** [comment]
Another way to use duckdb from php is via the php odbc client and duckdb's odbc support

**Comment by aszenz** [comment]
There is now a PHP ffi library to use duckdb:

https://github.com/satur-io/duckdb-php

**Reply by dhernandez** [reply]
Thanks for sharing.

You can also take a look at https://github.com/satur-io/duckdb-ext. Keep in mind this is my first PHP extension and my main focus is not there but in ffi library. Anyway I think it could be a good starting point for a DuckDB native extension.

Remember that contributions are welcome in both repositories 😄.
